### PR TITLE
Issue 1000: Configure Spring Boot Actuators

### LINKED ISSUE #1000: Configure Spring Boot Actuators
The code for this should be under the the [strongbox-web-core](https://github.com/strongbox/strongbox/tree/master/strongbox-web-core) module.

# Tasks:
* [ ] Configure Spring Boot Actuators
* [ ] Secure the Spring Boot Actuator end-points

# Useful links
* [Spring Boot: Actuators](https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html)
* [Baeldung: Spring Boot Actuators](https://www.baeldung.com/spring-boot-actuators)

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by adavid9**
fix #1000

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/config/SecurityConfig.java [hunk 1]
```diff
@@ -89,6 +90,10 @@ protected void configure(HttpSecurity http)
             // TODO SB-813
             .authenticationEntryPoint(customBasicAuthenticationEntryPoint())
             .and()
+            .authorizeRequests()
+            .requestMatchers(EndpointRequest.toAnyEndpoint())
+            .hasAuthority("ADMIN")
```
Comment (steve-todorov) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/config/SecurityConfig.java]: @ptirador shouldn't these be set under `application.properties` as `management.security.role=ADMIN` ?

**Comment by steve-todorov** [comment]
@carlspring should we add `[management.context-path](https://docs.spring.io/spring-boot/docs/1.0.0.RELEASE/reference/html/production-ready-monitoring.html#production-ready-customizing-management-server-context-path)` as well? (so the exposed endpoints are under `/api/monitoring` for example?)

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/config/SecurityConfig.java (hunk 1 above)
Comment (ptirador) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/config/SecurityConfig.java]: Yes, totally.

**Comment by carlspring** [comment]
@steve-todorov: Yes, I agree, we should!

File: strongbox-commons/src/main/resources/application.properties [hunk 2]
```diff
@@ -22,3 +22,9 @@ java.io.tmpdir=strongbox-vault/tmp
 ehcache.disk.store.dir=strongbox/cache
 
 spring.servlet.multipart.enabled=false
+
+management.endpoints.web.exposure.include=health,info,beans,metrics,trace
+management.endpoint.health.show-details=always
+management.endpoint.shutdown.enabled=true
+management.endpoints.web.base-path=/api/monitoring
+management.security.role=ADMIN
```
Comment (carlspring) [inline strongbox-commons/src/main/resources/application.properties]: Cosmetic: Could you please make sure that resource files such as these end with a new line (some consoles and editors could have an issue with this).

**Comment by carlspring** [comment]
@adavid9 ,

Could you please apply the changes for the end-points as well? Thanks! :)

**Comment by carlspring** [comment]
Thanks, @adavid9 ! :)

**Pull request merged**